docs/3_guides/2.2_modify_zkquiz_questions.md

```diff
@@ -0,0 +1,62 @@
+# Modify ZkQuiz Questions
+
+In [Build your first Aligned Application](2_build_your_first_aligned_application.md), we show how to build a trivia application, called ZkQuiz. ZKQuiz asks the user three questions, and if answered correctly, generates a ZK Proof of the correct answers, posts the proof on Aligned, and upon verification, mints an NFT via a smart contract.
+
+In this guide, we will show you how to replace those questions with others.
+
+{% hint style="warning" %}
+This guide assumes you have already read [Build your first Aligned Application](2_build_your_first_aligned_application.md)
+{% endhint %}
+
+## 1. Modify the Questions Asked
+
+First, we need to modify the questions presented to the user. To do this, navigate to our example in `examples/zkquiz/quiz/script/src/main.rs` and change the questions as needed:
+
+```Rust
+let mut user_answers = "".to_string();
+
+let question1 = "What is the capital of France?";
+let answers1 = ["Berlin", "Madrid", "Paris"];
+user_answers.push(ask_question(question1, &answers1));
+
+let question2 = "What is the chemical symbol for gold?";
+let answers2 = ["Au", "Ag", "Fe"];
+user_answers.push(ask_question(question2, &answers2));
+
+let question3 = "What is the native cryptocurrency of Ethereum?";
+let answers3 = ["Bitcoin", "Ether", "Litecoin"];
+user_answers.push(ask_question(question3, &answers3));
+```
+
+## 2. Update the Program
+
+Now we need to update the program to be proven with the new correct answers. As we saw in [Build your first Aligned Application](2_build_your_first_aligned_application.md), the program in `examples/zkquiz/quiz/program/src/main.rs` takes the user answers as inputs and checks that the hash of these inputs matches the expected output. Therefore, we need to update the expected output with the hash of our new correct answers. The updated example looks as follows:
+
+```Rust
+if output
+        != [
+            216, 11, 193, 177, 136, 178, 254, 150, 59, 128, 97, 103, 97, 128, 55, 57, 22, 242, 26,
+            1, 186, 223, 215, 118, 206, 47, 12, 206, 114, 118, 220, 93,
+        ]
+    {
+        panic!("Answers do not match");
+    }
+```
+
+## 3. Compile the Program
+
+Now we need to compile the updated Program, generating the binary file that will be run by the zkVm (ELF).
+For that you can use the [SP1 Rust toolchain](https://docs.succinct.xyz/introduction.html).
+
+```
+cd examples/zkquiz/quiz/program
+cargo prove build
+```
```

Comment:
I would add this as a Makefile target and explain that SP1 would be needed to run it.

Reply:
Done!